app/controllers/passkit/api/v1/registrations_controller.rb

```diff
@@ -105,6 +107,14 @@ def push_token
           json_body = JSON.parse(request.body.read)
           json_body["pushToken"]
         end
+
+        def push_service_url
+          return unless request&.body
+
+          request.body.rewind
```

Comment:
This looks fairly atypical. Can you not acccess it with `params["pushServiceUrl"]`?